### PR TITLE
Change demo app to use otel library and otlp exporter

### PR DESCRIPTION
**Pull request by bogdandrutu**
/cc @MrAlias this is an example of switching an opencensus app to opentelemetry, some facts: 50% more code, some TODOs left that will get fixed hopefully soon.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: examples/demo/app/main.go [hunk 1]
```diff
@@ -23,81 +23,132 @@ import (
...
+	otel.SetTextMapPropagator(propagation.TraceContext{})
+	otel.SetTracerProvider(tracerProvider)
+	otel.SetMeterProvider(pusher.MeterProvider())
+	pusher.Start()
+
+	return func() {
+		handleErr(tracerProvider.Shutdown(ctx), "failed to shutdown provider")
+		handleErr(exp.Shutdown(ctx), "failed to stop exporter")
+		pusher.Stop() // pushes any last exports to the receiver
 	}
```
Comment (tigrannajaryan) [inline examples/demo/app/main.go:92]: @MrAlias can we have a helper in Otel-go that makes this a single call, to which you only pass the exporter and the resource attributes? The decisions that this code makes appear to be reasonable defaults that majority of users will want to make (I may be wrong). Unless we want to force them to make this decisions consciously a helper would be very useful.

**Review by tigrannajaryan — APPROVED** [review]

File: examples/demo/app/main.go (hunk 1 above)
Comment (MrAlias) [reply examples/demo/app/main.go:92]: Yeah seems reasonable. We do that with other exporters (i.e. https://github.com/open-telemetry/opentelemetry-go/blob/fd3c82bec5fc8012da1fdd0402149b38bd8ec59d/exporters/stdout/exporter.go#L85-L93), looks like we haven't done it yet with OTLP.

Comment (MrAlias) [reply examples/demo/app/main.go:92]: Tracking with https://github.com/open-telemetry/opentelemetry-go/issues/1347

Comment (bogdandrutu) [reply examples/demo/app/main.go:92]: Thanks

**Pull request merged**